sympy/matrices/expressions/tests/test_matmul.py

```diff
@@ -44,53 +44,61 @@ def test_factor_in_front():
     assert factor_in_front(MatMul(A, 2, B, evaluate=False)) ==\
                            MatMul(2, A, B, evaluate=False)
 
+
 def test_remove_ids():
     assert remove_ids(MatMul(A, Identity(m), B, evaluate=False)) == \
                       MatMul(A, B, evaluate=False)
     assert null_safe(remove_ids)(MatMul(Identity(n), evaluate=False)) == \
                                  MatMul(Identity(n), evaluate=False)
 
+
 def test_xxinv():
     assert xxinv(MatMul(D, Inverse(D), D, evaluate=False)) == \
                  MatMul(Identity(n), D, evaluate=False)
 
+
 def test_any_zeros():
     assert any_zeros(MatMul(A, ZeroMatrix(m, k), evaluate=False)) == \
                      ZeroMatrix(n, k)
 
+
 def test_unpack():
     assert unpack(MatMul(A, evaluate=False)) == A
     x = MatMul(A, B)
     assert unpack(x) == x
 
-def test_only_squares():
-    A = MatrixSymbol('A', n, m)
-    C = MatrixSymbol('C', n, n)
-    D = MatrixSymbol('D', n, n)
 
+def test_only_squares():
     assert only_squares(C) == [C]
     assert only_squares(C, D) == [C, D]
     assert only_squares(C, A, A.T, D) == [C, A*A.T, D]
 
-def test_determinant():
-    A = MatrixSymbol('A', n, m)
-    C = MatrixSymbol('C', n, n)
-    D = MatrixSymbol('D', n, n)
 
+def test_determinant():
     assert det(2*C) == 2**n*det(C)
     assert det(2*C*D) == 2**n*det(C)*det(D)
     assert det(3*C*A*A.T*D) == 3**n*det(C)*det(A*A.T)*det(D)
 
-def test_doit():
-    C = MatrixSymbol('C', n, n)
-    D = MatrixSymbol('D', n, n)
 
+def test_doit():
     assert MatMul(C, 2, D).args == (C, 2, D)
```

Comment:
Out of curiousity, do you know why doesn't `MatMul` automatically canonicalize, like `Mul` does?